p2p/src/authenticated/lookup/mod.rs

```diff
@@ -251,101 +266,118 @@ mod tests {
                             // Add to received set
                             received.insert(sender);
                         }
-                    });
-
-                    // Send identity to all peers
-                    match mode {
-                        Mode::One => {
-                            for (j, (pub_key, _)) in peers.iter().enumerate() {
-                                // Don't send message to self
-                                if i == j {
-                                    continue;
-                                }
+                        complete_sender.send(()).await.unwrap();
 
-                                // Loop until success
-                                loop {
-                                    let sent = sender
-                                        .send(
-                                            Recipients::One(pub_key.clone()),
-                                            public_key.to_vec().into(),
-                                            true,
-                                        )
-                                        .await
-                                        .unwrap();
-                                    if sent.len() != 1 {
-                                        context.sleep(Duration::from_millis(100)).await;
-                                        continue;
-                                    }
-                                    assert_eq!(&sent[0], pub_key);
-                                    break;
-                                }
-                            }
+                        // Process messages until all finished (or else sender loops could get stuck as a peer may drop)
```

Comment:
The big tweak in this PR is to loop forever on re-sending messages (outside of the deterministic runtime, there is no way to guarantee delivery).